### PR TITLE
Enable standard via eslint and fix all linting issues.

### PR DESCRIPTION
**Pull request by haiodo**
Sorry folks, it is very huge, but fixes all linting issues. Tests works and UI is also works.

Signed-off-by: Andrey Sobolev <haiodo@gmail.com>

### CODE REVIEW

File: plugins/platform-ui/src/routes.ts [hunk 1]
```diff
@@ -313,18 +309,31 @@ export class Router<T> implements ApplicationRouter<T> {
...
+  /**
+   * Will return last one defined fragment.
+   */
+  private calcFragment (vars: Partial<T>, parents: Array<Router<any>>, children: Array<Router<any>>): string {
     let variables = this.variables as T
-    if (vars) {
-      variables = { ...variables, ...vars }
+    variables = { ...variables, ...vars }
+    // Check all children in reverse order
+    for (const r of children.reverse()) {
```
Comment (sumboid) [inline plugins/platform-ui/src/routes.ts:319]: I'm not sure about details, but just want to notice that `reverse` is mutable and that's probably not what we want from `calcFragment`.

File: plugins/platform-ui/src/routes.ts [hunk 2]
```diff
@@ -112,22 +112,22 @@ export class Router<T> implements ApplicationRouter<T> {
   ) {
     this.pattern = pattern
     this.parentRouter = parent
-    if (defaults) {
+    if (defaults != null) {
```
Comment (sumboid) [inline plugins/platform-ui/src/routes.ts:115]: I suppose it should be `!==` and `undefined`. There are a lot of same issues within file.

File: plugins/presentation/src/plugin.ts [hunk 3]
```diff
@@ -91,11 +91,11 @@ export default (platform: Platform, deps: { core: CoreService, i18n: I18n }): Pr
         const uxAttribute = model.as(attribute, ui.mixin.UXAttribute)
 
         const label = await i18nService.translate(uxAttribute.label)
-        const placeholder = uxAttribute.placeholder ? await i18nService.translate(uxAttribute.placeholder) : label
+        const placeholder = (uxAttribute.placeholder != null) ? await i18nService.translate(uxAttribute.placeholder) : label
```
Comment (sumboid) [inline plugins/presentation/src/plugin.ts:94]: seems to be `undefined` as well as presenter below

File: plugins/workbench/src/components/internal/DefaultPerspective.svelte [hunk 4]
```diff
@@ -107,7 +107,7 @@
 
           // Try find a class to be sure it is available.
           try {
-            (await coreService).getModel().getClass(_class as Ref<Class<Doc>>)
+            ;(await coreService).getModel().getClass(_class as Ref<Class<Doc>>)
```
Comment (sumboid) [inline plugins/workbench/src/components/internal/DefaultPerspective.svelte:110]: Unnecessary `;`

File: server/accounts/src/index.ts [hunk 5]
```diff
@@ -100,27 +100,27 @@ async function createAccount (db: Db, email: string, password: string): Promise<
 async function updateAccount (db: Db, email: string, password: string, newPassword: string, secondFactorEnabled: boolean, clientSecret: string, secondFactorCode: string): Promise<AccountInfo> {
   let account = await getAccount(db, email)
 
-  if (!account) {
+  if (account == null) {
```
Comment (sumboid) [inline server/accounts/src/index.ts:103]: `===` here and below

**Review by sumboid — COMMENTED** [review]
BTW with regard to `equal`, probably we need to add `eqeqeq` rule if it is not enabled.

File: dev/tool/.eslintrc.js [hunk 6]
```diff
@@ -0,0 +1 @@
+../../.eslintrc-global.js
```
[the remark this answers is not in the document]
Comment (haiodo) [reply dev/tool/.eslintrc.js:1]: Thanks Fixed., I've decided to use extends to make possible to some excludes, and extra configuration on individual basis.

File: dev/tool/src/index.ts [hunk 7]
```diff
@@ -77,24 +77,24 @@ program
...
-    withDatabase(mongodbUri, (client) => {
+  .action(async (workspace, cmd) => {
+    return await withDatabase(mongodbUri, (client) => {
       const accounts = client.db('accounts')
       const workspaceId = createWorkspace(accounts, workspace, cmd.organization)
 
       const tenant = withTenant(client, workspace)
       const databaseDone = initDatabase(tenant)
-      return Promise.all([workspaceId, databaseDone])
+      await Promise.all([workspaceId, databaseDone])
```
[the remark this answers is not in the document]
Comment (haiodo) [reply dev/tool/src/index.ts:87]: Fixed.

File: plugins/activity/src/details.ts [hunk 8]
```diff
@@ -1,22 +1,22 @@
...
-  if (details.name) {
+  if (details.name !== undefined) {
     if (result.length > 0) {
       result += ' - '
     }
     result += details.name
   }
   if (result.length === 0) {
-    return null
+    return undefined
```
[the remark this answers is not in the document]
Comment (haiodo) [reply plugins/activity/src/details.ts:19]: Nice, used.

File: plugins/platform-ui/src/routes.ts (hunk 1 above)
Comment (haiodo) [reply plugins/platform-ui/src/routes.ts:319]: Thanks, reverse mutation was unexpected by me :-(

File: plugins/platform-ui/src/routes.ts (hunk 2 above)
Comment (haiodo) [reply plugins/platform-ui/src/routes.ts:115]: Thanks. fixed.

File: plugins/presentation/src/plugin.ts (hunk 3 above)
Comment (haiodo) [reply plugins/presentation/src/plugin.ts:94]: Fixed. thanks.

File: plugins/workbench/src/components/internal/DefaultPerspective.svelte (hunk 4 above)
Comment (haiodo) [reply plugins/workbench/src/components/internal/DefaultPerspective.svelte:110]: Fixed.

File: server/accounts/src/index.ts (hunk 5 above)
Comment (haiodo) [reply server/accounts/src/index.ts:103]: Nope, getAccount() return `| null` so check is correct here. it uses mongo API and it return null.

File: plugins/activity/src/details.ts [hunk 9]
```diff
@@ -1,22 +1,22 @@
...
-  if (!details) {
-    return null
+export function getTransactionObjectDetails (tx: ObjectTx): string | undefined {
+  const details = tx._txDetails
+  if (details === undefined) {
+    return undefined
   }
   let result = ''
-  if (details.id) {
+  if (details.id !== undefined) {
```
[the remark this answers is not in the document]
Comment (haiodo) [reply plugins/activity/src/details.ts:9]: Fixed.

File: dev/tool/src/index.ts [hunk 10]
```diff
@@ -41,19 +41,19 @@ program
...
   .requiredOption('-w, --workspace <workspace>', 'workspace')
-  .action((email, cmd) => {
-    withDatabase(mongodbUri, async (client) => {
+  .action(async (email, cmd) => {
+    return await withDatabase(mongodbUri, async (client) => {
       const db = client.db('accounts')
 
       // Create user account inside accounts
-      const user = await getUserAccount(db, email).then(user => !user ? createUserAccount(db, email, cmd.password) : Promise.resolve(user))
+      const user = await getUserAccount(db, email).then(user => user == null ? await createUserAccount(db, email, cmd.password) : Promise.resolve(user))
```
[the remark this answers is not in the document]
Comment (haiodo) [reply dev/tool/src/index.ts:49]: Fixed.

File: server/accounts/src/index.ts (hunk 5 above)
Comment (sumboid) [reply server/accounts/src/index.ts:103]: Yeah, but is it intentional to use `==` here?

Comment (haiodo) [reply server/accounts/src/index.ts:103]: Fixed.

**Review by z2sx — APPROVED** [review]

**Comment by qwabra** [comment]
Replace from `!= null` to `!== null` look like mistake `(undefined == null) == true`. I think `!= null`  means checking for `undefined`

**Comment by qwabra** [comment]
that strange too
```ts
export interface EditorActions {
  toggleBold (): void // is replaced to
  toggleBold: () => void
}
```

------

```ts
interface A {
    a(): void
    b: () => void
}
class B {
    a() { }
    b = () => { }
}

declare var a: A
declare var b: B
```

https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgILIN4FgBQz-JwAUAlAFzIBuA9sACa4HIBGFpyAvAHxW0M4BfXAgA2cAM7jkAIUyMCxEpmRC8BZp2Ttuy1atx0IouFBSUThCqgNGxpqhdYzcQA

**Comment by haiodo** [comment]
Hi @qwabra 

1. Based on code it could be defined or null , it should not be undefined, so !== null should be fine.
https://eslint.org/docs/rules/eqeqeq
2. EditorActions are replaced according to: 
https://github.com/typescript-eslint/typescript-eslint/blob/v4.22.0/packages/eslint-plugin/docs/rules/method-signature-style.md

**Comment by z2sx** [comment]
Hey @qwabra! Wanna join project? we offer good compensation :)

**Comment by qwabra** [comment]
@haiodo
1. if u say so "Based on code".
2. hmmm. some voodo magic, some ?broken voodo magic [TSPlay](https://www.typescriptlang.org/play?strict=true#code/PQKhFgCgAIWhTAHgQwLYAcA29oDMBOA9qtACaEDGAzgFxSzQAWALs+rcMAO48B0zAT3TwqFfAEt0zTMgB2Ac16F884OWrBGc0gCNChANbB88bMirwAtLMLMRwQcNESplgEyWAbLxapMAYipmCQpmS1wAV1lQ8UJZS0cRehBgKABuKChxWTt8XGQKHABBWXFUZExoAG9oAF8snPg8gpwAEUJ5BEQ7WVIqaBKyiuq6htz8wugAYWRmLp6+gdLyypr6yChSeAoZE2hsOdwARhpoAApEU8GVgEpoAF4APmgAN0JxUgzILZ3kPYO8G5ThdTu15Hcnq93p9NttdjgAbgAMzAy7TWYQ55vD5fKDHB6AtLQTjQACi+CI+GgXHEzEY0EsliCIWYADEojE4gAVIRJSC4NwE45EkkAeQMeMF9zwSJFwDJFOUUCAA)

@z2sx , check email plz

**Pull request merged**